### DISCUSSION TITLE
Cardinalities affect on match weight

### DISCUSSION THREAD

**Discussion by nadd0u** [moj-analytical-services/splink#1980] [Q&A]
I am building a model that identifies records belonging to the same person using the following fields: 
* First name 
* Last Name
* Address 
Each of these fields has some ambiguity (variations in spelling). The ambiguity is more dramatic in the name fields than in the address fields. 

My issue is that my model is barely penalizing mismatches in first name. As a result, as long as two records belong to the same family (matching last name + address) it will consider them a match. 

I suspect this is because the first name field has the lowest cardinality. There is a decent chance that two non matches will share the same first name and little chance that two non matches will share the same address. 

Does this diagnosis seem correct, if so - is there something i can do to combat this?

**Comment by samkodes** [comment]
I have recently been struggling with similar issues. 
Can you share your estimated M-probabilities, U-probabilities, and match weights?
A small negative match weight on first name may be the result of a too-high "all others" m-probability for first name, which in turn may be caused by dirty data with insufficient flexibility in match levels (e.g. fuzzy matching). Adding more flexibility in match levels may help in this case.
Another option may be to try to use TF adjustments on your address field. 
You always also have the freedom to manually exclude mismatches on FN in post-processing (via filtering on match pattern), or to adjust your match threshold to exclude the match pattern.
